src/client/builtin_scene/text/sdf/atlas.hpp

```diff
@@ -0,0 +1,175 @@
+#pragma once
+
+#include <vector>
+#include <memory>
+#include <unordered_map>
+#include <cstdint>
+#include "./tiny_sdf.hpp"
+
+namespace builtin_scene::text::sdf
+{
+  /**
+   * UV coordinates for a glyph in an atlas
+   */
+  struct GlyphUV
+  {
+    float u0, v0, u1, v1; // UV coordinates in atlas texture
+    int x, y;             // Pixel coordinates in atlas
+    int width, height;    // Glyph dimensions
+
+    GlyphUV() = default;
+    GlyphUV(float u0, float v0, float u1, float v1, int x, int y, int w, int h)
+        : u0(u0)
+        , v0(v0)
+        , u1(u1)
+        , v1(v1)
+        , x(x)
+        , y(y)
+        , width(w)
+        , height(h)
+    {
+    }
+  };
+
+  /**
+   * A texture atlas containing SDF glyphs
+   */
+  class SDFAtlas
```

Comment:
SDF still use the instance texture altas, please remove these classes in this file.